node/src/node/builder.rs

```diff
@@ -0,0 +1,309 @@
+            keypair: Some(keypair),
+            ..self
+        }
+    }
+
+    /// Set the bootstrap nodes to connect and trust.
+    ///
+    /// **Default:** [`Network::canonical_bootnodes`]
+    pub fn bootnodes<I>(self, addrs: I) -> Self
+    where
+        I: IntoIterator<Item = Multiaddr>,
+    {
+        NodeBuilder {
+            bootnodes: addrs.into_iter().collect(),
+            ..self
+        }
+    }
+
+    /// Set the addresses where [`Node`] will listen for incoming connections.
+    pub fn listen<I>(self, addrs: I) -> Self
+    where
+        I: IntoIterator<Item = Multiaddr>,
+    {
+        NodeBuilder {
+            listen: addrs.into_iter().collect(),
+            ..self
+        }
+    }
+
+    /// Maximum number of headers in batch while syncing.
+    ///
+    /// **Default:** 512
+    pub fn sync_batch_size(self, batch_size: u64) -> Self {
+        NodeBuilder {
+            sync_batch_size: Some(batch_size),
+            ..self
+        }
+    }
+
+    /// Set syncing window.
+    ///
+    /// Syncing window defines maximum age of a block considered for syncing and sampling.
+    ///
+    /// **Default if [`InMemoryStore`]/[`InMemoryBlockstore`] are used:** 60 seconds.\
+    /// **Default:** 30 days.\
+    /// **Minimum:** 60 seconds.
+    pub fn syncing_window(self, dur: Duration) -> Self {
+        NodeBuilder {
+            syncing_window: Some(dur),
+            ..self
+        }
+    }
+
+    /// Set pruning delay.
+    ///
+    /// Pruning delay how much time the pruner should wait after syncing window in
+    /// order to prune the block.
+    ///
+    /// **Default if [`InMemoryStore`]/[`InMemoryBlockstore`] are used:** 60 seconds.\
+    /// **Default:** 1 hour.\
+    /// **Minimum:** 60 seconds.
+    pub fn pruning_delay(self, dur: Duration) -> Self {
+        NodeBuilder {
+            pruning_delay: Some(dur),
+            ..self
+        }
+    }
+
+    fn build_config(self) -> Result<NodeConfig<B, S>, NodeBuilderError> {
+        let network = self.network.ok_or(NodeBuilderError::NetworkNotSpecified)?;
+
+        let bootnodes = if self.bootnodes.is_empty() {
+            network.canonical_bootnodes().collect()
+        } else {
+            self.bootnodes
+        };
+
+        if bootnodes.is_empty() && self.listen.is_empty() {
+            // It is a valid scenario for user to create a node without any bootnodes
+            // and listening addresses. However it may not be what they wanted. Because
+            // of that we display a warning.
+            warn!("Node has empty bootnodes and listening addresses. It will never connect to another peer.");
+        }
+
+        // `Node` is memory hungry when in-memory stores are used and the user most
+        // probably will never know that they need to have a smaller syncing window
+        // to avoid it. In order to make `lumina` more user friendly we decided to
+        // have a different default when our in-memory stores are used.
```

Comment:
```suggestion
        // `Node` is memory hungry when in-memory stores are used and the user may not
        // expect they should set a smaller syncing window to reduce that. For user-friendliness
        // sake, use smaller default syncing window, if we're running in memory.
```
Maybe? Reworded for conciseness.